src/Map/Map.js

```diff
@@ -60,6 +61,9 @@ class Map extends React.Component {
     // setup select interactions for the map
     this.initializeSelect(this.map)
 
+    // callback for <Provider> if it's not mounted
+    this.props.setContextProps(this.getContextProps())
```

Comment:
called with the map uid, set the relevant context props (`map`, `selectInteraction`, etc.)

Reply:
Provider mounts, Map mounts (with a prop of `ProviderContext`), on mount, Map creates `map`, `selectInteraction`, (attaches those to `ProviderContext`/callback?), Provider re-renders map with these two new props?

a `maps` prop to `<Provider>` would avoid this^